Mathlib/CategoryTheory/Generator.lean

```diff
@@ -567,4 +569,98 @@ theorem wellPowered_of_isDetector [HasPullbacks C] (G : C) (hG : IsDetector G) :
   haveI := small_subsingleton ({G} : Set C)
   wellPowered_of_isDetecting hG
 
+theorem wellPowered_of_isSeparator [HasPullbacks C] [Balanced C] (G : C) (hG : IsSeparator G) :
+    WellPowered C := wellPowered_of_isDetecting hG.isDetector
+
+section HasGenerator
+
+variable (E : Type u₃) [Category.{v₃} E]
+
+/--
+A category instantiates `HasSeparator` if and only if it has a separator.
```

Comment:
This docstring sounds a bit strange to me, it almost makes it seem like there is some magic metaprogram that automatically instantiates `HasSeparator`.

You could consider going into a bit more detail that `HasSeparator` is `Prop`-valued, so users who care about specific objects being separators should use `IsSeparator` instead.